src/components/canvas-tools/text-tool.tsx

```diff
@@ -0,0 +1,95 @@
+import * as React from "react";
+import memoize from "memoize-one";
+import { Editor } from "slate-react";
+import Plain from "slate-plain-serializer";
+import { ToolTileModelType } from "../../models/tools/tool-tile";
+import { TextContentModelType } from "../../models/tools/text/text-content";
+import { Value } from "slate";
+
+import "./text-tool.sass";
+​
+interface IProps {
+  model: ToolTileModelType;
+  readOnly?: boolean;
+}
+
+interface IState {
+  value: Value;
+}
+
+export default class TextToolComponent extends React.Component<IProps, IState> {
+  // Set the initial value when the app is first constructed.
+  public state = {
+    value: Plain.deserialize("")
+  };
+
+  private editor: Editor | null;
+
+  // use memoization to only deserialize when contents change
+  // cf. https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization
+  private deserializeTextContent = memoize(
```

Comment:
Wouldn't this work better as a view in the model?  Mobx would handle broadcasting updates for you when the underlying model changed so you wouldn't need this `memoize` function.